test/integration/features/experimental-flags.feature

```diff
@@ -0,0 +1,62 @@
+@experimental-flags
+Feature: Experimental Flags
+  Experimental flag --extra-clusterup-flags will be enabled by setting MINISHIFT_ENABLE_EXPERIMENTAL environment variable,
+  this flag will provide access to some of upcoming feature and experiments.
+
+  Scenario: User cannot start minishift experimental feature directly
+    Given Minishift has state "Does Not Exist"
+     Then executing "minishift start --service-catalog" fails
```

Comment:
What is this scenario testing.. because you can not run `minishift start --service-catalog`... as right now there is no flag `--service-catalog` for `minishift start` command, experimental or otherwise..

Reply:
I am not  convinced that this a required check.. as `--service-catalog` is passed as argument to `--extra-clusterup-flags`. and is not meant to be passed to `minishift start` command.

> As a user i am going to start experimental feature without knowing the environment setting. This way we can validate the feature functionality such as in which condition it should fail or pass

Are you concerned that people are going to use the `--service-catalog` flag directly and we should prevent them? Then this is something for the docs to make clear.

Reply:
I understand your concern 
> I am not convinced that this a required check.. as --service-catalog is passed as argument to --extra-clusterup-flags. and is not meant to be passed to minishift start command.

This is a check nothing much. If user passes it alone it should fail, right ? 

> Are you concerned that people are going to use the --service-catalog flag directly and we should prevent them? Then this is something for the docs to make clear

This is just a abstract form the doc and putting them to the feature file scenario. Mostly scenario contains actual and negative test cases. Theses negative test case comprises wrong data  and can be anything (not necessarily what is exactly there in the doc).

Reply:
> This is a check nothing much. If user passes it alone it should fail, right ?

Right..

>This is just a abstract form the doc and putting them to the feature file scenario. Mostly scenario contains actual and negative test cases. Theses negative test case comprises wrong data and can be anything (not necessarily what is exactly there in the doc).

I get that you are trying to make sure passing that flag to minishift makes it fail to start, but in that case you should use an experimental flag which can be directly passed to minishift, like the `--no-provision` flag.

The appropriate checks for testing the `--service-catalog` flag is there in `Scenario: User cannot use minishift experimental flag --extra-clusterup-flag without experimental feature name`.

Might be i am missing something, lets see what @minishift/minishift-dev thinks.

Reply:
I feel you are mixing negative scenario step with actual scenario step. Anyway lets wait feedback from @minishift/minishift-dev

Reply:
It is not a critical test, but I am fine with it. It is an usual practice to mix positive and negative tests in testing in general.